src/solvers/flattening/bv_pointers.cpp

```diff
@@ -45,7 +45,7 @@ literalt bv_pointerst::convert_rest(const exprt &expr)
       }
     }
   }
-  else if(expr.id()==ID_dynamic_object)
+  else if(expr.id()==ID_is_dynamic_object)
   {
     if(operands.size()==1 &&
```

Comment:
These should probably become DATA_INVARIANTS but this is a clean up and not vital for this PR.

Reply:
These being the condition `operands.size() == 1 && operands[0].type().id() == ID_pointer`?

Reply:
Yes.  They should be enforced on exprt creation with PRECONDITION, on conversion with DATA_INVARIANT (as you have done in std_expr) and then referenced / checked with DATA_INVARIANT when they are relevant for the usage (for example here).